src/compiler/emitter.ts

```diff
@@ -1881,13 +1860,77 @@ var __param = (this && this.__param) || function (paramIndex, decorator) {
                 }
             }
 
+            function emitCallOrNewExpressionTarget(node: Expression): CallTarget {

```

Comment:
I just found a better way to do this. We can just call it `emitCallTarget()` and return the target and have a separate function for the emit of accessor or just inline the emit logic in `emitNewExpression` since we only use it there. 

The problem is though, that it already exists one function with `emitCallTarget()`. Any chance I can just rename it to `emitCallTargetWithOptionalTempAssignment()`?

```typescript
function emitCallTarget(node: Expression): Expression {
    if (node.kind === SyntaxKind.Identifier || node.kind === SyntaxKind.ThisKeyword || node.kind === SyntaxKind.SuperKeyword) {
        emit(node);
        return node;
    }
    let temp = createAndRecordTempVariable(TempFlags.Auto);

    write("(");
    emit(temp);
    write(" = ");
    emit(node);
    write(")");
    return temp;
}
```